### PR TITLE
Do not send migrate task when BE only have 1 storage medium

### LINKED ISSUE #4000: A large number of useless migration tasks cause the locks of FE and BE to take a long time
### Steps to reproduce the behavior (Required)
1. create a ssd table & default storage medium set to HDD & BE only config HDD
2. create SSD table e.g.
CREATE TABLE `student` (
  `name` varchar(100) NOT NULL COMMENT "",
  `weight` int(11) NOT NULL COMMENT "",
  `height` decimal64(9, 3) NULL COMMENT ""
) ENGINE=OLAP 
DUPLICATE KEY(`name`)
COMMENT "student"
DISTRIBUTED BY HASH(`name`) BUCKETS 100
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"storage_medium" = "SSD"
);

### Expected behavior (Required)
Normal
### Real behavior (Required)
There are lot of MIGRATE TASK Failed and send again and again.
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4000

## Problem Summary(Required) ：

when be report tablet info to fe. 
If we know that this BE must have no other medium, then only write a log as a reminder, not to send a task.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -182,8 +185,16 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
                                 long partitionId = tabletMeta.getPartitionId();
                                 TStorageMedium storageMedium = storageMediumMap.get(partitionId);
                                 if (storageMedium != null && backendTabletInfo.isSetStorage_medium()) {
+                                    // If storage medium is less than 1, there is no need to send migration tasks to BE.
+                                    // Because BE will ignore this request.
                                     if (storageMedium != backendTabletInfo.getStorage_medium()) {
-                                        tabletMigrationMap.put(storageMedium, tabletId);
+                                        if (backendStorageTypeCnt <= 1) {
+                                            LOG.info("available storage medium type count is less than 1, " +
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:192]: I think debug level is enough

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 2]
```diff
@@ -128,6 +128,9 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
             }
         }
 
+        long backendStorageTypeCnt = Catalog.getCurrentSystemInfo().getBackend(backendId).getDisks().values()
+                .stream().map(u -> u.getStorageMedium().getValue()).distinct().count();
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:132]: I want to know if it always has no nullptr exception

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 14 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|0|14|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 3]
```diff
@@ -128,6 +131,18 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
             }
         }
 
+        int backendStorageTypeCnt = -1;
+        Backend be = Catalog.getCurrentSystemInfo().getBackend(backendId);
+        if (be != null) {
+            ImmutableMap<String, DiskInfo> disks = be.getDisks();
+            Set<Integer> set = Sets.newHashSet();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:138]: Better to use Set\<TStorageMedium\>.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 10 / 15 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|10|15|66.67%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 4]
```diff
@@ -269,6 +283,21 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
                 tabletMigrationMap.size(), transactionsToClear.size(), transactionsToPublish.size(), (end - start));
     }
 
+    @VisibleForTesting
+    public int getBackendStorageTypeCnt(long backendId) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:287]: Better to move to Backend.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 9 / 17 (52.94%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|0|8|00.00%|
|:large_blue_circle:|com/starrocks/system/Backend.java|9|9|100.00%|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**